reports-scheduler/src/main/java/com/amazon/opendistroforelasticsearch/reportsscheduler/metrics/RollingCounter.java

```diff
@@ -0,0 +1,102 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.reportsscheduler.metrics;
+
+import java.time.Clock;
+import java.util.concurrent.ConcurrentSkipListMap;
+import java.util.concurrent.atomic.LongAdder;
+
+/**
+ * Rolling counter. The count is refreshed every interval. In every interval the count is cumulative.
+ */
+public class RollingCounter implements Counter<Long> {
+    private static final long METRICS_ROLLING_WINDOW_VALUE = 3600L;
+    private static final long METRICS_ROLLING_INTERVAL_VALUE = 60L;
+
+    private final long capacity;
+    private final long window;
+    private final long interval;
+    private final Clock clock;
+    private final ConcurrentSkipListMap<Long, Long> time2CountWin;
+    private final LongAdder count;
+
+    public RollingCounter() {
+        this(METRICS_ROLLING_WINDOW_VALUE, METRICS_ROLLING_INTERVAL_VALUE);
+    }
+
+    public RollingCounter(long window, long interval, Clock clock) {
+        this.window = window;
+        this.interval = interval;
+        this.clock = clock;
+        time2CountWin = new ConcurrentSkipListMap<>();
+        count = new LongAdder();
+        capacity = window / interval * 2;
+    }
+
+    public RollingCounter(long window, long interval) {
+        this(window, interval, Clock.systemDefaultZone());
+    }
+
+    @Override
+    public void increment() {
+        add(1L);
+    }
+
+    @Override
+    public void add(long n) {
+        trim();
+        time2CountWin.compute(getKey(clock.millis()), (k, v) -> (v == null) ? n : v + n);
```

Comment:
may be use read-friendly variable names?

Reply:
Renamed from `time2CountWin` to `timeToCountMap `